### ISSUE TITLE
Error installing openssl on CentOS7

### ISSUE THREAD

**Issue by ericmjl** [Homebrew/linuxbrew-core#16644]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [ ] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link? **NOTE: no output from `brew gist-logs git`**
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
    - [brew config](https://gist.github.com/ericmjl/8d0da07e266c018acff3a63315f26425)
    - [brew doctor](https://gist.github.com/ericmjl/abff9196ec697fdb17cb3a6942ae74df)

To help us debug your issue please explain:

- What you were trying to do (and why)

I was trying to install `git` to replace the system version (which was old).

- What happened (include command output)

I first ran:

```
brew install git
```

The failure happened when installing openssl 1.1.1d as part of the dependency chain of `git`. I get the following error (embedded in the `details` tab):

<details>

```
$ brew install git
==> Installing dependencies for git: openssl@1.1, libbsd and expat
==> Installing git dependency: openssl@1.1
==> Downloading https://www.openssl.org/source/openssl-1.1.1d.tar.gz
Already downloaded: /home/me/.cache/Homebrew/downloads/567481d93e12a0cdaf990a51159f5ccf67fdb7bbcf6a1d961df071365e44731e--openssl-1.1.1d.tar.gz
==> perl ./Configure --prefix=/home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d --openssldir=/home/me/.linuxbrew/etc/openssl@1.1 no-ssl3 no-ssl3-method no-zlib  enable-md2 linux-x86_64
==> make
==> make install MANDIR=/home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/share/man MANSUFFIX=ssl
Last 15 lines from /home/me/.cache/Homebrew/Logs/openssl@1.1/03.make:
install ./include/openssl/uierr.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/uierr.h
install ./include/openssl/whrlpool.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/whrlpool.h
install ./include/openssl/x509.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/x509.h
install ./include/openssl/x509_vfy.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/x509_vfy.h
install ./include/openssl/x509err.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/x509err.h
install ./include/openssl/x509v3.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/x509v3.h
install ./include/openssl/x509v3err.h -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/include/openssl/x509v3err.h
install libcrypto.a -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.a
install libssl.a -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libssl.a
link /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so.1.1
install libcrypto.so -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so
install libcrypto.so.1.1 -> /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so
ar: /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so: file format not recognized
ar: /home/me/.linuxbrew/Cellar/openssl@1.1/1.1.1d/lib/libcrypto.so.new: file format not recognized
make: *** [install_dev] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
openssl@1.1: Enable building on ARM https://github.com/Homebrew/linuxbrew-core/pull/16642
```

</details>

- Step-by-step reproduction instructions (by running `brew install` commands)

```bash
# After installing a fresh copy of linuxbrew:
$ brew install git
```

**Comment by dawidd6** [comment]
Maybe listing `binutils` as a build dependency would help?

**Comment by ericmjl** [comment]
I resolved this issue by figuring a number of key steps:

1. Firstly, the `perl` that linuxbrew was using to install openssl was using was an old version bundled with the system. Test command for figuring this out came from [this comment](https://github.com/openssl/openssl/issues/6641#issuecomment-4035706680) on the OpenSSL issue tracker.
2. Secondly, I had to edit the Brew formula, such that `ENV["PERL"]` pointed to a different path that was more up-to-date. 
3. After updating the formula, installation went smoothly. I took the chance to install another `perl` directly into homebrew _just in case_.
